cmd/neofs-cli/internal/client/client.go

```diff
@@ -384,19 +477,60 @@ func (x GetObjectRes) Header() *object.Object {
 //
 // Returns any error prevented the operation from completing correctly in error return.
 // For raw reading, returns *object.SplitInfoError error if object is virtual.
-func GetObject(prm GetObjectPrm) (res GetObjectRes, err error) {
-	var getPrm client.GetObjectParams
+func GetObject(prm GetObjectPrm) (*GetObjectRes, error) {
+	var getPrm client.PrmObjectGet
 
-	getPrm.WithAddress(prm.objAddr)
-	getPrm.WithPayloadWriter(prm.wrt)
-	getPrm.WithRawFlag(prm.raw)
+	if id := prm.objAddr.ContainerID(); id != nil {
+		getPrm.FromContainer(*id)
+	}
 
-	res.cliRes, err = prm.cli.GetObject(context.Background(), &getPrm, append(prm.opts,
-		client.WithSession(prm.sessionToken),
-		client.WithBearer(prm.bearerToken),
-	)...)
+	if id := prm.objAddr.ObjectID(); id != nil {
+		getPrm.ByID(*id)
+	}
 
-	return
+	if prm.sessionToken != nil {
+		getPrm.WithinSession(*prm.sessionToken)
+	}
+
+	if prm.bearerToken != nil {
+		getPrm.WithBearerToken(*prm.bearerToken)
+	}
+
+	if prm.raw {
+		getPrm.MarkRaw()
+	}
+
+	if prm.local {
+		getPrm.MarkLocal()
+	}
+
+	rdr, err := prm.cli.ObjectGetInit(context.Background(), getPrm)
+	if err != nil {
+		return nil, fmt.Errorf("init object reading on client: %w", err)
+	}
+
+	var hdr object.Object
+
+	if !rdr.ReadHeader(&hdr) {
+		_, err = rdr.Close()
+		return nil, fmt.Errorf("read object header: %w", err)
+	}
+
+	sz := hdr.PayloadSize()
+	if sz > 1024 {
```

Comment:
Where does this constant comes from?

Reply:
From the assumption that NeoFS CLI cannot afford large data buffers. I'll create a const and add docs.